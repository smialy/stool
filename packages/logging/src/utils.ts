import { Levels } from './consts';

const NAMES = Object.keys(Levels);

export function levelToMask(level: number) {
    return level === Levels.NOTSET ? Levels.NOTSET : level * 2 - 1;
}

export function checkLevel(level: number | string): number {
    const type = typeof level;
    if (type === 'number') {
        return level as number;
    } else if (type === 'string') {
        level = (level as string).toUpperCase();
        if (NAMES.includes('' + level)) {
            return Levels[level] as number;
        }
    }
    const msg = `Level is not number or valid string: "${level}" [${NAMES}]`;
    throw TypeError(msg);
}

export function isException(ex: any) {
    return !!(ex && ex.message && ex.stack);
}
