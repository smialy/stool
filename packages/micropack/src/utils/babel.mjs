import babel from '@babel/core';
import babelPlugin from '@rollup/plugin-babel';

export async function createBabelConfig(options) {
    const plugins = await createBabelPlugins('plugin', [
        {
            name: 'babel-plugin-transform-async-to-promises',
            inlineHelpers: true,
            externalHelpers: false,
            minify: true,
        },
        {
            name: '@babel/plugin-transform-react-jsx',
            pragma: options.jsx || 'h',
            // pragmaFrag: customOptions.pragmaFrag || 'Fragment',
        },
        {
            name: '@babel/plugin-proposal-class-properties',
            loose: true,
        },
        {
            name: '@babel/plugin-transform-regenerator',
            async: false,
        },
    ]);
    const presets = await createBabelPlugins('preset', [
        {
            name: '@babel/preset-env',
            loose: true,
            useBuiltIns: false,

        }
    ])
    return babelPlugin.getBabelOutputPlugin({
        plugins,
        presets,
    });
}

async function createBabelPlugins(type, configs) {
    return Promise.all(configs.map(async ({ name, ...options}) => {
        const module = await import(name);
        return babel.createConfigItem([module.default, options], { type });
    }));
}