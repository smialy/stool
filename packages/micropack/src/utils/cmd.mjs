export const collectDict = (vals, acc) => ({
    ...acc,
    ...vals
        .split(',')
        .map((item) => item.split(':'))
        .reduce((acc, val) => {
            acc[val[0]] = val[1];
            return acc;
        }, {}),
});
export const collectList = (vals, acc=[]) => {
    return [...acc, ...vals.split(',')];
}
export const increaseVerbose = (_, acc) => (acc += 1);
