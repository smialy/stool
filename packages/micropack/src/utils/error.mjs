function logError(error) {
    if (error && error.stack) {
        error = error.stack;
    }
    let message = `\n${error} \n\nNode.js ${process.version}\n`;
    console.log(message);
}

export function setupExceptionHandler() {
    process.on("uncaughtException", logError);
    process.on('unhandledRejection', logError);  // catch all promisess
}
