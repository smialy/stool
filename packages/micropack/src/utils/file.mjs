import fs from 'fs';
import { promisify } from 'util';

export const readFile = promisify(fs.readFile);
export const readFileStat = promisify(fs.stat);

export function isDir(filePath) {
    stat(filePath)
        .then((stat) => stat.isDirectory())
        .catch(() => false);
}

export function isFile(filePath) {
    stat(filePath)
        .then((stats) => stats.isFile())
        .catch(() => false);
}

export async function readJsonFile(filePath) {
    const payload = await readFile(filePath, 'utf8');
    try {
        return JSON.parse(payload);
    } catch (e) {
        console.warn(e);
    }
}
