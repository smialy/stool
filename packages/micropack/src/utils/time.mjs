export function elapsed() {
    const start = new Date().getTime();
    return () => {
        const end = new Date().getTime();
        let time = end - start;
        let sufix = 'ms';
        if (time >= 1000) {
            time = time / 1000;
            sufix = 'sec';
        }
        return `${time}${sufix}`;
    };
}
