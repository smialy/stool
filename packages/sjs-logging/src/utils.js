import {LEVELS} from './consts';


const NAMES = Object.keys(LEVELS);

/**
 * @param {nuber} level Log level
 * @return {number}
 */
export function levelToMask(level) {
    return level === LEVELS.NOTSET ? LEVELS.NOTSET : level * 2 - 1;
}

/**
 *
 * @param {numbe|string} level
 */
export function checkLevel(level){
    let type = typeof level;
    if(type === 'number'){
        return level;
    }else if(type === 'string'){
        if(NAMES.includes(level)){
            return LEVELS[level];
        }
    }
    let msg = `Level not number or valid string: "${level}" [${NAMES}]`;
    throw TypeError(msg);
}