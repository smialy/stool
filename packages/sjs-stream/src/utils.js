export const noop = () => { };
export function toObserver(onData, onError, onComplete) {
    if (typeof onData === 'function') {
        return {
            onData,
            onError: onError || noop,
            onComplete: onComplete || noop
        };
    }
    return Object.assign({
        onData: noop,
        onError: noop,
        onComplete: noop
    }, onData || {});
}
export function runMethod(method, ...args) {
    if (typeof method === 'function') {
        try {
            method(...args);
        }
        catch (e) {
            console.error(e);
        }
    }
}
export function enqueue(fn) {
    Promise.resolve().then(() => {
        try {
            fn();
        }
        catch (e) {
            console.error(e);
        }
    });
}
