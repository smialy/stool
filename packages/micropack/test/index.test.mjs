import { readdirSync, statSync, readFileSync } from 'fs';
import url from 'node:url';
import path from 'node:path';
import chai, { expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';

import { buildFixture, printDirTree, findAllFiles } from './utils.mjs';

chai.use(jestSnapshotPlugin());


const __dirname = url.fileURLToPath(new URL('.', import.meta.url));

const FIXTURES_DIR = `${__dirname}fixtures`;


describe('fixtures', () => {
    const files = readdirSync(FIXTURES_DIR);
    const dirs = files
        .map(file => path.join(FIXTURES_DIR, file))
        .filter(file => statSync(file).isDirectory());

    dirs.forEach((fdir) => {
        const rdir = path.relative(__dirname, fdir);
        it(`build ${rdir}`, () => {
            const output = buildFixture(fdir);
            const tree = printDirTree(fdir);
            expect([
                "Tree:",
                tree,
                "Output:",
                output,
                ].join('\n'),
            ).toMatchSnapshot();

            const distDir = path.join(fdir, 'dist');
            const files = findAllFiles(distDir, entry => {
                if (/.map$/.test(entry.name)) return false;
                return true;
            });
            for(const [name, filePath] of files) {
                expect(readFileSync(filePath, { encoding: 'utf8'})).toMatchSnapshot(`dist/${name}`);
            }
        });
    });
});
