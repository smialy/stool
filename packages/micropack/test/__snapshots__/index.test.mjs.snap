// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures build fixtures/compress 1`] = `
"Tree:
compress
  dist
    index.js
    index.js.map
    index.mjs
    index.mjs.map
  package.json
  src
    foo.ts
    index.ts

Output:
"
`;

exports[`fixtures build fixtures/compress 2`] = `
"function foobar(){return\\"foobar\\"}export{foobar};
//# sourceMappingURL=index.mjs.map
"
`;

exports[`fixtures build fixtures/compress 3`] = `
"exports.foobar=function foobar(){return\\"foobar\\"};
//# sourceMappingURL=index.js.map
"
`;

exports[`fixtures build fixtures/css-module 1`] = `
"Tree:
css-module
  dist
    index.js
    index.mjs
  package.json
  src
    declarations.d.ts
    index.ts
    main.css
    main.module.css

Output:
"
`;

exports[`fixtures build fixtures/css-module 2`] = `
"var css_248z = \\".foo {\\\\n    background: #f3f3f3;\\\\n}\\\\n\\";

var scoped = {\\"foo\\":\\"main-module_foo__alNSp\\"};

const index = () => {
    const el1 = document.createElement('div');
    el1.innerHTML = css_248z;
    const el2 = document.createElement('div');
    el2.className = scoped.foo;
};

export { index };
"
`;

exports[`fixtures build fixtures/css-module 3`] = `
"'use strict';

var css_248z = \\".foo {\\\\n    background: #f3f3f3;\\\\n}\\\\n\\";

var scoped = {\\"foo\\":\\"main-module_foo__alNSp\\"};

const index = () => {
    const el1 = document.createElement('div');
    el1.innerHTML = css_248z;
    const el2 = document.createElement('div');
    el2.className = scoped.foo;
};

exports.index = index;
"
`;

exports[`fixtures build fixtures/no-modern 1`] = `
"Tree:
no-modern
  dist
    index.js
    index.js.map
    index.mjs
    index.mjs.map
  package.json
  src
    foo.ts
    index.ts

Output:
"
`;

exports[`fixtures build fixtures/no-modern 2`] = `
"const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

export { foobar };
//# sourceMappingURL=index.mjs.map
"
`;

exports[`fixtures build fixtures/no-modern 3`] = `
"'use strict';

const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

exports.foobar = foobar;
//# sourceMappingURL=index.js.map
"
`;

exports[`fixtures build fixtures/no-sourcemap 1`] = `
"Tree:
no-sourcemap
  dist
    index.js
    index.mjs
  package.json
  src
    foo.ts
    index.ts

Output:
"
`;

exports[`fixtures build fixtures/no-sourcemap 2`] = `
"const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

export { foobar };
"
`;

exports[`fixtures build fixtures/no-sourcemap 3`] = `
"'use strict';

const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

exports.foobar = foobar;
"
`;

exports[`fixtures build fixtures/simple 1`] = `
"Tree:
simple
  dist
    index.js
    index.js.map
    index.mjs
    index.mjs.map
  package.json
  src
    foo.ts
    index.ts

Output:
"
`;

exports[`fixtures build fixtures/simple 2`] = `
"const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

export { foobar };
//# sourceMappingURL=index.mjs.map
"
`;

exports[`fixtures build fixtures/simple 3`] = `
"'use strict';

const foo = () => 'foo';

function foobar() {
    return foo() + 'bar';
}

exports.foobar = foobar;
//# sourceMappingURL=index.js.map
"
`;
